### PR TITLE
provide decrypt and encrypt as adapter methods

### PR DESCRIPTION
**Pull request by foxriver76**
- and add according tests
- we use a system.config.native.secret in tests to keep it more realistic

### CODE REVIEW

**Comment by Apollon77** [comment]
Will check tomorrow

**Comment by Apollon77** [comment]
Thank you!

File: test/lib/testAdapterHelpers.js [hunk 1]
```diff
@@ -378,6 +378,21 @@ function register(it, expect, context) {
...
+        expect(sysConfig).to.be.ok;
+        const secret = sysConfig.native.secret;
+        expect(secret).to.be.ok;
+        const encrypted = context.adapter.encrypt(secret, 'topSecret');
+        // we only check not equal, because encryption can change and is not always deterministic
+        expect(encrypted).to.not.equal('topSecret');
+        const decrypted = context.adapter.decrypt(secret, encrypted);
+        // check that correctly decrypted
+        expect(decrypted).to.equal('topSecret');
+        return Promise.resolve();
```
Comment (AlCalzone) [inline test/lib/testAdapterHelpers.js:393]: FYI: This line is not necessary.

**Comment by AlCalzone** [comment]
Do we need a feature flag for this?

**Comment by Apollon77** [comment]
No Feature flag because methods can be checked for existence.

**Pull request merged**